File: LeetCode/src/929.cpp
// Unique Email Addresses
#include <iostream>
#include <string.h>
#include <unordered_set>
#include <vector>
using namespace std;

class Solution {
public:
  int numUniqueEmails(vector<string> &emails) {
    unordered_set<string> uniqueEmails;
    for (string &email : emails) {
      string cleanEmail;
      for (char c : email) {
        if (c == '+' || c == '@') {
          break;
        }
        if (c == '.') {
          continue;
        }
        cleanEmail += c;
      }
      cleanEmail += email.substr(email.find('@'));
      uniqueEmails.insert(cleanEmail);
    }
    return uniqueEmails.size();
  }
};

File: LeetCode/src/929.js
// Unique Email Addresses
const numUniqueEmails = function (emails) {
	const uniqueEmails = [];
	emails.forEach((email) => {
		const part = email.split("@");
		part[0] = part[0].replace(/\./g, "");
		if (part[0].includes("+")) {
			part[0] = part[0].substring(0, part[0].indexOf("+"));
		}
		const cleanEmail = part.join("@");
		if (!uniqueEmails.includes(cleanEmail)) {
			uniqueEmails.push(cleanEmail);
		}
	});
	return uniqueEmails.length;
};
